### PR TITLE
Inline standing approvals with action configurations

### LINKED ISSUE #929: Inline standing approvals into action configurations
## Problem

Standing approvals and action configurations are conceptually the same thing — "what can this agent do, and under what constraints?" — but the UI treats them as completely separate entities on different pages with different creation flows. This forces users to manage the same concept in two places.

The current action config already has an approval mode (auto-approve vs requires-approval). Standing approvals are just the "auto-approve" case with optional execution limits and expiration. That should be a property *on* the config, not a separate page.

## Goal

Make "standing approval" a property of an action configuration rather than a separate first-class UI concept. Every standing approval must have a backing action configuration — standalone standing approvals are no longer supported.

### Connector Config Page (`/agents/:agentId/connectors/:connectorId`)
- Add a **"Standing Approval" column** to the action configurations table showing status (active / none / expired / exhausted)
- Clicking the status or a toggle opens an inline panel or small dialog to configure standing approval fields (max executions, expiration)
- Remove the need to navigate away to manage standing approvals

### Dashboard
- Replace the current standalone Standing Approvals card/table with a **summary view** that shows all action configs with active standing approvals across all connectors
- This becomes a read-only "quick view" — clicking a row navigates to the connector config page where the actual management happens

### Data Model
- Standing approvals **must** have a `source_action_configuration_id` — no more standalone standing approvals
- Backend API can remain mostly unchanged (standing approvals stay as a separate resource linked via `source_action_configuration_id`), but the frontend treats them as a property of the config

### Phase 1: Connector Config Page — Add Standing Approval Column
- [ ] Add "Standing Approval" column to the action configurations table in `ActionConfigurationsSection.tsx`
- [ ] Show standing approval status per config (active/none/expired/exhausted) using `useLinkedStandingApprovalsForConfig` hook
- [ ] Add inline UI (toggle or small dialog) to create/edit/revoke a standing approval directly from the action config row
- [ ] Include standing approval fields: max executions, expiration date

### Phase 2: Dashboard — Replace Standalone Standing Approvals Section
- [ ] Replace `StandingApprovalsCard` with a summary/filtered view of action configs that have active standing approvals
- [ ] Each row links back to the connector config page for management
- [ ] Remove the standalone `CreateStandingApprovalDialog` multi-step flow from the dashboard

### Phase 3: Enforce Action Config Requirement
- [ ] Update backend to require `source_action_configuration_id` on standing approval creation (or create one automatically)
- [ ] Migrate any existing standalone standing approvals to have backing action configs
- [ ] Remove standalone standing approval creation paths from the API

### Manual Testing
- [ ] Verify creating a standing approval from the connector config page works end-to-end
- [ ] Verify the dashboard summary view correctly shows all active standing approvals across connectors
- [ ] Verify revoking/editing standing approvals from the connector config page
- [ ] Verify expiration and execution limit behavior unchanged
- [ ] Test on mobile viewport (the screenshots show mobile usage)

### PR DESCRIPTION
**Pull request by chiedocursor**
## Summary

Implements [GitHub issue #929](https://github.com/supersuit-tech/permission-slip/issues/929): standing approvals are treated as a property of an action configuration in the UI, and creation now always ties to a backing `source_action_configuration_id`.

### Backend / data

- `POST /standing-approvals/create` requires `source_action_configuration_id`, validates it exists, belongs to the same agent, and matches `action_type`.
- Migration backfills NULL/dangling `source_action_configuration_id` values, then sets `NOT NULL` and adds an FK to `action_configurations(id)` with `ON DELETE RESTRICT`.
- **Action config delete:** before deleting a configuration, all standing approvals referencing that config are removed so the FK does not block deletion when non-active rows exist (replaces revoke-only cleanup).
- Seed data and test fixtures updated for the new constraint.
- OpenAPI: `CreateStandingApprovalRequest` and `StandingApproval` schema updates (bundled spec regenerated).

### Frontend

- Connector **Action Configurations** table: new **Standing Approval** column with status badge; sheet to create/edit limits or revoke (uses config parameters for constraints).
- Dashboard **Standing Approvals** card: read-only summary of active linked approvals with **Manage** links to the connector config page (removed standalone create flow from the card).
- `CreateStandingApprovalDialog`: optional `initialSourceActionConfigurationId` for future wiring; create path guards missing config id for TypeScript/API alignment.
- Query invalidation extended so connector rows refresh after standing-approval mutations.

Closes #929

### Developer

- [x] `make bundle` and `npm run generate:api` (frontend) after OpenAPI edits
- [x] `npx tsc --noEmit` in `frontend/`
- [x] Vitest: `StandingApprovalsCard`, `CreateStandingApprovalDialog`, `ReviewApprovalDialog`, `ConnectorConfigPage`
- [ ] `make test-backend` (Go tests not executed here: local `go` toolchain could not satisfy `cloud.google.com/go/bigquery` Go version requirement in this environment)

### Manual Testing

- [ ] Create standing approval from connector config sheet (non-wildcard config) and confirm API success
- [ ] Dashboard summary lists only active linked approvals; **Manage** navigates to correct connector page
- [ ] Revoke and re-create from connector page; limits/expiry updates behave as before
- [ ] Delete an action configuration that had historical (non-active) standing approvals and confirm delete succeeds
- [ ] Mobile viewport on connector table + sheet
- [ ] Run full `make test` / `make build` in a CI-capable environment

<div><a href="https://cursor.com/agents/bc-c0ca878c-29cf-4bc4-b20a-a927c2464610"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-web-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-web-light.png">Open in Web</picture></a>&nbsp;<a href="https://cursor.com/background-agent?bcId=bc-c0ca878c-29cf-4bc4-b20a-a927c2464610"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-cursor-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-cursor-light.png">Open in Cursor</picture></a>&nbsp;</div>

### CODE REVIEW

**Comment by chiedocursor** [comment]
## 🤖 Watch Session Summary

### Changes Made
No changes made during this session.

### Decision Log

#### ✅ Implemented
No review comments acted on.

---
*Watch session ended after 0 minutes of inactivity. Processed 0 comments across 1 poll cycles.*

File: frontend/src/hooks/useStandingApprovalsForConfigs.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    queryFn: async () => {
+      if (!accessToken) throw new Error("Missing access token");
+      const all: StandingApproval[] = [];
+      let after: string | undefined;
+      for (;;) {
+        const { data, error } = await client.GET("/v1/standing-approvals", {
+          headers: { Authorization: `Bearer ${accessToken}` },
+          params: {
+            query: {
+              status: "all",
```
Comment (chiedoclaude) [inline frontend/src/hooks/useStandingApprovalsForConfigs.ts]: **[High · Performance]** This hook fetches **all** standing approvals for the user (`status: "all"`) with full pagination, then filters client-side by config ID. The API already supports a `source_action_configuration_id` query filter (added in this very PR's OpenAPI changes) — but this hook doesn't use it.

**Why:** For users with many standing approvals (across multiple agents/connectors), this fetches potentially hundreds of records when only a handful are needed per config. As the user base grows, this becomes an unbounded client-side data transfer.

**Fix:** Either:
1. Make one query per config ID using the `source_action_configuration_id` filter (simplest, N queries for N configs):
```ts
const results = await Promise.all(
  sortedIds.map(async (configId) => {
    const { data } = await client.GET("/v1/standing-approvals", {
      headers: { Authorization: `Bearer ${accessToken}` },
      params: { query: { source_action_configuration_id: configId, status: "all" } },
    });
    return { configId, approvals: data?.data ?? [] };
  })
);
```
2. Or add a batch endpoint that accepts multiple config IDs and returns grouped results.

Option 1 is a quick win — each query returns only the relevant rows, avoiding the full scan + client-side filter.

File: db/standing_approvals.go [hunk 2]
```diff
@@ -325,6 +325,21 @@ func RevokeActiveStandingApprovalsForSourceActionConfig(ctx context.Context, db
 	return tag.RowsAffected(), nil
 }
 
+// DeleteStandingApprovalsForSourceActionConfig deletes all standing approval rows
+// that reference the given action configuration. Used when removing an action
+// configuration so FK constraints (when present) are satisfied.
+func DeleteStandingApprovalsForSourceActionConfig(ctx context.Context, db DBTX, userID, sourceConfigID string) (int64, error) {
+	tag, err := db.Exec(ctx,
+		`DELETE FROM standing_approvals
```
Comment (chiedoclaude) [inline db/standing_approvals.go:338]: **[Medium · Data Integrity]** This `DELETE` removes **all** standing approval rows for the config — active, expired, revoked, and exhausted. The previous code (`RevokeActiveStandingApprovalsForSourceActionConfig`) only changed active rows to `revoked`, preserving historical records.

**Why:** Deleting an action configuration now permanently destroys the entire audit trail of standing approvals that referenced it. If someone needs to audit past approvals (e.g., "which actions were auto-approved last month?"), the records are gone.

**Context:** I understand this is necessary because the new `ON DELETE RESTRICT` FK + `NOT NULL` constraint means you can't just revoke — you must remove **all** references to delete the config. This is a valid design trade-off.

**Suggestion:** Consider a soft-delete or tombstone approach as a future improvement: instead of hard-deleting the action config, set `status = 'deleted'` (allowing the FK references to remain intact). For now, a code comment documenting this intentional data loss trade-off would help future maintainers understand why `DELETE` was chosen over `UPDATE ... SET status = 'revoked'`.

File: db/migrations/20260413194737_standing_approvals_require_source_action_config.sql [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+-- Remaining rows: create one backing disabled config per standing approval.
+DO $$
+DECLARE
+    r RECORD;
+    new_id text;
+    resolved_connector text;
+    fallback_connector text;
+BEGIN
+    SELECT id INTO fallback_connector FROM connectors ORDER BY id LIMIT 1;
```
Comment (chiedoclaude) [inline db/migrations/20260413194737_standing_approvals_require_source_action_config.sql:41]: **[Medium · Correctness]** If the `connectors` table is empty, `fallback_connector` will be `NULL`. Later, when `resolved_connector` falls through to `fallback_connector` (line 59), the `INSERT INTO action_configurations` at line 67 will fail because `connector_id` is `NOT NULL`.

**Why:** An empty connectors table is unlikely in production, but possible in fresh environments or during testing. A migration that can fail partway through leaves the schema in an inconsistent state.

**Fix:** Add a guard after the fallback query:
```sql
IF fallback_connector IS NULL THEN
    RAISE NOTICE 'No connectors exist — skipping standing approval backfill (no orphan rows can reference a valid connector).';
    RETURN;
END IF;
```
Alternatively, if standing approvals can't exist without connectors in practice, add an assertion:
```sql
IF fallback_connector IS NULL AND EXISTS (SELECT 1 FROM standing_approvals WHERE source_action_configuration_id IS NULL) THEN
    RAISE EXCEPTION 'Cannot backfill standing approvals: no connectors exist';
END IF;
```

**Review by chiedoclaude — COMMENTED** [review]
## Review Round 1/6

**Findings:** 3 (0 critical, 1 high, 2 medium, 0 low)

The core change — making `source_action_configuration_id` required with a FK constraint — is well-designed and the migration is thoughtfully structured. The main concern is the `useStandingApprovalsForConfigs` hook which fetches **all** standing approvals for the user and filters client-side, rather than using the server-side `source_action_configuration_id` filter that was added in this very PR. The migration also has a minor edge case if the connectors table is empty.

**Note on bundled spec:** `frontend/src/api/bundled.yaml` includes ~500 lines of unrelated changes (PayPal connector docs, Dropbox PKCE, channels/users endpoints, template bulk-apply). These appear to be from a full spec regeneration that picked up unreleased changes from other branches. This bloats the diff and increases merge conflict risk — consider regenerating from only the standing approval schema changes if possible.

### What Looks Good
- The data model change is sound — every standing approval now references a backing action config, making the relationship explicit and queryable
- Migration is well-structured: link existing rows to matching configs, create disabled backing configs for orphans, then enforce the constraint
- Server-side validation is comprehensive: checks existence, agent ownership, and action type match before allowing creation
- Test coverage is thorough — new tests for missing config ID, wrong agent, and all existing tests properly updated with backing configs
- The dashboard card is now a read-only summary with "Manage" links, cleanly separating the overview from the management UI on the connector page
- Query invalidation across create/update/revoke hooks is comprehensive
- Fixture helpers (`ensureStandingApprovalSourceConfig`) are well-factored — the basic `InsertStandingApproval` delegates to `InsertStandingApprovalWithStatus` which creates the backing config, so all existing tests keep working

---
*Automated review by `/review` — Round 1 of 6*

File: api/standing_approvals.go [hunk 4]
```diff
@@ -237,23 +237,46 @@ func handleCreateStandingApproval(deps *Deps) http.HandlerFunc {
...
+
 		var constraintsBytes []byte
-		if req.SourceActionConfigurationID != nil && len(req.Constraints) > 0 {
+		if len(req.Constraints) > 0 {
 			s := strings.TrimSpace(string(req.Constraints))
 			if s == "{}" || s == "null" {
 				// Match-all parameters for this action type (trusted when tied to a source config,
 				// e.g. template customize with all-wildcard parameters). Stored as NULL in DB.
 				constraintsBytes = nil
 			} else {
```
Comment (chiedoclaude) [inline api/standing_approvals.go:282]: **[High · Correctness]** The condition was simplified from `if req.SourceActionConfigurationID != nil && len(req.Constraints) > 0` to just `if len(req.Constraints) > 0`. Since `source_action_configuration_id` is now always required, this is **logically correct** — but two existing tests still expect the old behavior and will fail:

1. `TestCreateStandingApproval_ConstraintsEmptyObject` — sends `constraints: {}`, expects 400. Now `len("{}") > 0` → enters this branch → `s == "{}"` → treated as match-all → **creation succeeds (201)**.
2. `TestCreateStandingApproval_ConstraintsNull` — sends `constraints: null`, expects 400. `json.RawMessage` holds the literal bytes `null` (len=4) → enters this branch → `s == "null"` → treated as match-all → **creation succeeds (201)**.

**Why this matters:** Backend tests were not run in this PR (`make test-backend` skipped per the PR description). These two tests will fail in CI.

**Fix:** If `{}` and `null` constraints are now intentionally allowed (which aligns with the PR's stated design — "trusted flows may send `{}` to mean match all"), update the tests:
```go
// TestCreateStandingApproval_ConstraintsEmptyObject
// {} is now valid (match-all) when tied to a source action config
if w.Code != http.StatusCreated {
    t.Fatalf("expected 201 for match-all constraints, got %d: %s", w.Code, w.Body.String())
}
```
And update the test comments to explain the new semantics. If `{}` and `null` should still be rejected, restore the old guard:
```go
if req.SourceActionConfigurationID != nil && len(req.Constraints) > 0 {
```
(But since the field is now always non-nil, this would effectively be `if len(req.Constraints) > 0` anyway — so the tests are the ones that need updating.)

**Review by chiedoclaude — COMMENTED** [review]
## Review Round 2/6

**Findings:** 1 (0 critical, 1 high, 0 medium, 0 low)

### Changes Since Last Round
No new commits since Round 1. Re-examining the constraint validation logic with fresh eyes revealed a test/code inconsistency that will cause backend test failures.

### Remaining Issues
1. **[High · Correctness]** Constraint validation for `{}` and `null` has silently changed behavior — two tests will fail (see inline comment)
2. Prior Round 1 findings (performance hook, migration fallback, delete history) remain unaddressed

### What Looks Good
- The validation rewrite is *logically correct* for the new data model (every SA has a backing config, so match-all via `{}` is reasonable)
- The constraint handling for non-trivial cases (omitted, all-wildcard, mixed, too-large, non-object) is unaffected and will pass

---
*Automated review by `/review` — Round 2 of 6*

**Comment by chiedocursor** [comment]
**Re: automated review (Round 1)** — Addressed the high-priority hook issue (per-config API queries), added migration guard for empty `connectors`, and documented the standing-approval hard-delete trade-off on action config removal. The `bundled.yaml` noise note is acknowledged: regenerating only touched paths would shrink future diffs; for this PR the bundle stays aligned with `make bundle` output.

File: frontend/src/hooks/useStandingApprovalsForConfigs.ts (hunk 1 above)
Comment (chiedocursor) [reply frontend/src/hooks/useStandingApprovalsForConfigs.ts]: Implemented: `useStandingApprovalsForConfigs` now uses `useQueries` with `source_action_configuration_id` + pagination per config ID (pushed to this branch).

File: db/standing_approvals.go (hunk 2 above)
Comment (chiedocursor) [reply db/standing_approvals.go:338]: Documented the trade-off: hard DELETE is required for FK + NOT NULL; comments on the DB helper and delete handler. Soft-delete on configs would be a good follow-up.

File: db/migrations/20260413194737_standing_approvals_require_source_action_config.sql (hunk 3 above)
Comment (chiedocursor) [reply db/migrations/20260413194737_standing_approvals_require_source_action_config.sql:41]: Migration guard: if orphan standing approvals remain and `connectors` is empty, migration raises a clear exception instead of failing mid-loop on NULL `connector_id`.

File: api/standing_approvals.go (hunk 4 above)
Comment (chiedocursor) [reply api/standing_approvals.go:282]: Updated `TestCreateStandingApproval_ConstraintsEmptyObject` and `_ConstraintsNull` to expect **201** when `source_action_configuration_id` is set — `{}` / JSON `null` are match-all (same as create path). Pushed in commit on this branch.

File: frontend/src/hooks/useStandingApprovalsForConfigs.ts (hunk 1 above)
Comment (chiedocursor) [reply frontend/src/hooks/useStandingApprovalsForConfigs.ts]: Already addressed in latest commits (per-config queries / migration guard / delete documentation).

File: db/standing_approvals.go (hunk 2 above)
Comment (chiedocursor) [reply db/standing_approvals.go:338]: Already addressed in latest commits (per-config queries / migration guard / delete documentation).

File: db/migrations/20260413194737_standing_approvals_require_source_action_config.sql (hunk 3 above)
Comment (chiedocursor) [reply db/migrations/20260413194737_standing_approvals_require_source_action_config.sql:41]: Already addressed in latest commits (per-config queries / migration guard / delete documentation).

File: frontend/src/hooks/useStandingApprovalsForConfigs.ts [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+  );
+
+  const results = useQueries({
+    queries: sortedIds.map((configId) => ({
+      queryKey: ["standing-approvals-for-config", userId ?? "", configId],
+      queryFn: async () => {
+        const token = accessToken;
+        if (!token) throw new Error("Missing access token");
+        return fetchAllStandingApprovalsForConfig(token, configId);
+      },
```
Comment (chiedoclaude) [inline frontend/src/hooks/useStandingApprovalsForConfigs.ts:59]: **[Medium · Correctness]** The query key was renamed from `"standing-approvals-by-config-ids"` to `"standing-approvals-for-config"`, but all three mutation hooks (`useCreateStandingApproval`, `useRevokeStandingApproval`, `useUpdateStandingApproval`) still invalidate the old key:

```ts
queryClient.invalidateQueries({
  queryKey: ["standing-approvals-by-config-ids"],
});
```

**Why:** React Query's `invalidateQueries` uses prefix matching on the array. `"standing-approvals-by-config-ids"` does **not** prefix-match `"standing-approvals-for-config"` — they're different strings. After creating/updating/revoking a standing approval, the connector page's per-config queries won't be invalidated, showing stale data until manual refresh.

**Mitigating factor:** The `ActionConfigStandingApprovalSheet` calls `onSuccess()` → `refetchStanding()` which explicitly refetches. But mutations from the dashboard dialog (or any other path) won't refresh the connector page cache.

**Fix:** Update the invalidation in all three hooks:
```ts
queryClient.invalidateQueries({
  queryKey: ["standing-approvals-for-config"],
});
```
This prefix-matches all per-config queries regardless of userId/configId.

**Review by chiedoclaude — COMMENTED** [review]
## Review Round 3/6

**Findings:** 1 (0 critical, 0 high, 1 medium, 0 low)

### Changes Since Last Round
Two new commits addressing all prior findings:
- `50d6e8f` — Rewrote `useStandingApprovalsForConfigs` to use per-config server-side filtering (addresses High · Performance), added migration guard (addresses Medium · Correctness), added code comments documenting delete trade-off (addresses Medium · Data Integrity)
- `2951bad` — Updated `ConstraintsNull` and `ConstraintsEmptyObject` tests to expect 201 (addresses High · Correctness)

All prior issues are resolved. One new issue introduced by the rewrite.

### Remaining Issues
1. **[Medium · Correctness]** Query key mismatch — mutation hooks invalidate the old key, per-config queries won't refresh after mutations (see inline comment)

### What Looks Good
- The `useQueries` rewrite is clean and well-structured — per-config parallel queries with a `combine` function that preserves the same return interface
- Migration guard is exactly what was suggested — clear error message when connectors table is empty
- Test updates have good comments explaining the new match-all semantics
- Trade-off comments in Go code are helpful for future maintainers

---
*Automated review by `/review` — Round 3 of 6*

File: frontend/src/hooks/useStandingApprovalsForConfigs.ts (hunk 5 above)
Comment (chiedocursor) [reply frontend/src/hooks/useStandingApprovalsForConfigs.ts:59]: Fixed: create/update/revoke hooks now invalidate `["standing-approvals-for-config"]` to match `useStandingApprovalsForConfigs` query keys (pushed).

Comment (chiedocursor) [reply frontend/src/hooks/useStandingApprovalsForConfigs.ts:59]: Resolved — see prior reply + commit.

File: api/standing_approvals.go (hunk 4 above)
Comment (chiedocursor) [reply api/standing_approvals.go:282]: Tests updated in commit 2951bad — both cases now expect 201 with backing config. Thread can be resolved.

**Comment by chiedocursor** [comment]
## 🤖 Watch Session Summary

### Changes Made
No changes made during this session.

### Decision Log

#### ✅ Implemented
- chiedoclaude asked for Per-config standing approval queries → implemented in ``
- chiedoclaude asked for Document hard-delete trade-off → implemented in ``
- chiedoclaude asked for Migration empty connectors guard → implemented in ``
- chiedoclaude asked for Tests for {} and null constraints with source config → implemented in ``
- chiedoclaude asked for Review round 1 summary → implemented in ``
- chiedoclaude asked for Review round 2 summary → implemented in ``
- chiedocursor asked for PR follow-up comment → implemented in ``
- chiedoclaude asked for Mutation hooks invalidate standing-approvals-for-config query key → implemented in ``
- chiedoclaude asked for Review round 3 summary → implemented in ``

#### ❌ Declined
- chiedocursor suggested Stale watch wrap-up → declined because Prior automated session; superseded.

---
*Watch session ended after 0 minutes of inactivity. Processed 10 comments across 6 poll cycles.*

## 🔧 CI / audit remediation

### Remediation round 1

- **ci.yml** — conclusion `failure` — Goose split the DO $$ block on semicolons causing unterminated dollar quote. Wrapped DO block in StatementBegin/StatementEnd (commit 9c3364ce).

### Remediation round 2

- **ci.yml** — conclusion `failure` — Second CI: migration inserted connector on empty DB (connector tests); missing 90-day expiry check; delete test expected revoked row. Fixed in 48847693.

### Remediation round 3

- **ci.yml** — conclusion `failure` — Third CI: sentinel connector persisted in shared DB; db package unused ctx. Reverted sentinel insert, added cleanup migration 20260413211258, fixed tests (d9c2d58a).

### Remediation round 4

- **ci.yml** — conclusion `failure` — Fourth CI: cleanup migration did not delete sentinel (FK rows); frontend Dashboard test title. Deep-delete in 20260413211258 + Dashboard.test (08af11c4).

### Remediation round 5

- **ci.yml** — conclusion `failure` — Fifth CI: goose already applied 20260413211258 before deep-delete edit. Added new migration 20260413212758 (ed2fb7fc).

**Review by chiedoclaude — COMMENTED** [review]
## Review Round 4/6

No new issues found.

### Changes Since Last Round
- `136bd17` — Fixed React Query invalidation key in all three mutation hooks (`useCreateStandingApproval`, `useRevokeStandingApproval`, `useUpdateStandingApproval`): `"standing-approvals-by-config-ids"` → `"standing-approvals-for-config"`. This addresses the stale cache issue flagged in Round 3.

All prior comments have been addressed across 3 fix commits. No additional issues found on re-examination.

---
*Automated review by `/review` — Round 4 of 6*

**Review by chiedoclaude — COMMENTED** [review]
## Review Round 5/6

No new issues found.

### Changes Since Last Round
- `9c3364c` — Added `-- +goose StatementBegin` / `-- +goose StatementEnd` directives around the `DO $$` PL/pgSQL block in the migration. Without these, goose would split on semicolons inside the block, breaking the migration. Good catch.

All prior comments have been addressed. The PR is looking clean.

---
*Automated review by `/review` — Round 5 of 6*

**Comment by chiedoclaude** [comment]
## Code Review Complete

**PR:** #930 — Inline standing approvals with action configurations
**Author:** @chiedocursor
**Rounds completed:** 5/6
**Reason for stopping:** max turns reached (all issues resolved)

### Outstanding Issues
None — this PR looks good to merge after running `make test-backend` to verify the constraint test changes pass.

### Review History
| Round | Inline Comments | Notes |
|-------|----------------|-------|
| 1 | 3 | Initial review: performance (fetch-all hook), data integrity (delete history), migration fallback guard |
| 2 | 1 | Constraint validation behavior change — 2 Go tests expect 400 but would get 201 |
| 3 | 1 | Query key mismatch after `useQueries` rewrite — mutation hooks invalidating stale key |
| 4 | 0 | All prior issues fixed (query key fix in `136bd17`) |
| 5 | 0 | Goose migration fix (`9c3364c` — StatementBegin/End directives) |

### Overall Assessment
This is a well-executed data model change that makes the relationship between standing approvals and action configurations explicit and enforced. The migration handles legacy data thoughtfully (link existing, create backing configs for orphans, then enforce constraints).

The PR author was highly responsive — every review finding was addressed in follow-up commits within minutes:
- The `useStandingApprovalsForConfigs` hook was completely rewritten from a fetch-all-filter-client approach to per-config server-side queries using `useQueries`
- Go constraint tests were aligned with the new match-all semantics
- Migration edge case (empty connectors table) guarded
- React Query invalidation keys fixed
- Goose `StatementBegin`/`StatementEnd` directives added

**Remaining manual verification items:**
1. Run `make test-backend` in a CI-capable environment — Go tests were not executed in the original PR
2. Run `make build` to catch TypeScript compilation issues
3. Manual testing of the connector page standing approval column + dashboard summary card

---
*Review performed by Claude Code `/review` skill*
